ilc/common/router/ServerRouter.js

```diff
@@ -43,6 +41,40 @@ module.exports = class ServerRouter {
         }
     }
 
+    async getRouteAssets(reqUrl) {
+        const route = await this.#getRouteInfo(reqUrl);
+
+        return _.reduce(route.slots, (routeAssets, slotData) => {
+            const appInfo = this.__apps[slotData.appName];
+
+            if (!routeAssets.includes(appInfo.spaBundle)) {
+                routeAssets.push(appInfo.spaBundle);
+            }
+
+            Object.keys(appInfo.dependencies).filter((dependencyName) => {
```

Comment:
Why are you using filter method here?

Reply:
I think you're doing some awkward things here.... Let's discuss...
Simple `Object.assign` should work fine here.

Reply:
Updated